src/assets/js/protocol/protocol-modal.js

```diff
@@ -1 +1,159 @@
 // Protocol Modal - Hello World
+// create namespace
+if (typeof Mozilla === 'undefined') {
+    var Mozilla = {};
+}
+
+Mozilla.Modal = (function(w) {
+    'use strict';
+
+    var open = false;
+    var body = w.document.body;
+    var html = w.document.documentElement;
+    var options = {};
+    var pageContentParent;
+    var pageContent;
+    var modal;
+
+    /*
+    origin: element that triggered the modal
+    content: content to display in the modal
+    options: object of optional params:
+        title: title to display at the top of the modal
+        onCreate: function to fire after modal has been created
+        onDestroy: function to fire after modal has been closed
+        allowScroll: boolean - allow/restrict page scrolling when modal is open
+        closeText: string to use for close button a11y.
+    */
+    var _createModal = function(origin, content, opts) {
+        options = opts;
+
+        var isSmallViewport = window.innerWidth < 760;
+
+        // Make sure modal is closed (if one exists)
+        if (open) {
+            _closeModal();
+        }
+
+        // Create new modal
+        var title = (options && options.title) ? options.title : '';
+        var closeText = (options && options.closeText) ? options.closeText : '';
+
+        var modalFragment =
+            '<div class="mzp-c-modal" role="dialog" aria-labelledby="' + origin.getAttribute('id') + '" tabindex="-1">' +
+            '  <div class="mzp-c-modal-window">' +
+            '    <div class="mzp-c-modal-inner">' +
+            '      <header><h2>' + title + '</h2></header>' +
+            '      <div class="mzp-c-modal-close">' +
+            '        <button type="button" class="mzp-c-modal-button-close">' + closeText + '</button>' +
+            '      </div>' +
+            '    </div>' +
+            '  </div>' +
+            '</div>';
+
+        if ((options && !options.allowScroll) || isSmallViewport) {
+            html.classList.add('noscroll');
```

Comment:
Should this class follow the naming convention for states? So it would be `mzp-is-noscroll`